File: Remove_Edge.py
def remove_edge(src, dest):
    if src < 1 or src > V[0] or dest < 1 or dest > V[0]:
        print("Invalid vertices for edge removal.")
        return

    adj_list[src - 1] = [vtx for vtx in adj_list[src - 1] if vtx != dest]
    adj_list[dest - 1] = [vtx for vtx in adj_list[dest - 1] if vtx != src]
}

File: Remove_Edge.c
void removeEdge(struct Graph* graph, int src, int dest) {
    if (src < 1 || src > graph->V || dest < 1 || dest > graph->V) {
        fprintf(stderr, "Invalid vertices for edge removal.\n");
        return;
    }

    // Remove the edge from src to dest
    struct Node* prev = NULL;
    struct Node* curr = graph->adjList[src - 1];
    while (curr) {
        if (curr->data == dest) {
            if (prev) {
                prev->next = curr->next;
            } else {
                graph->adjList[src - 1] = curr->next;
            }
            free(curr);
            break;
        }
        prev = curr;
        curr = curr->next;
    }

    // Remove the edge from dest to src
    prev = NULL;
    curr = graph->adjList[dest - 1];
    while (curr) {
        if (curr->data == src) {
            if (prev) {
                prev->next = curr->next;
            } else {
                graph->adjList[dest - 1] = curr->next;
            }
            free(curr);
            break;
        }
        prev = curr;
        curr = curr->next;
    }
}
